pkg/utils/secrets/manager/manager.go

```diff
@@ -95,6 +95,15 @@ type (
 		dataChecksum               string
 		lastRotationInitiationTime int64
 	}
+
+	// Rotation specifies certain configuration options for the rotation of secrets.
+	Rotation struct {
```

Comment:
Maybe call this `Options` (similar to `manager.Options`/`client.Options`/...) or `Config` instead?